alf/algorithms/dynamic_action_repeat_agent.py

```diff
@@ -279,3 +319,39 @@ def summarize_train(self, experience, train_info, loss_info, params):
             field = alf.nest.find_field(train_info, 'action_distribution')
             if len(field) == 1:
                 summary_utils.summarize_action_dist(field[0])
+
+    def preprocess_experience(self, rl_exp):
+        """Normalize training rewards if a reward normalizer is provided. Shape
+        of ``rl_exp`` is ``[B, T, ...]``. The statistics of the normalizer is
+        updated by random sample rewards.
+        """
+        reward = rl_exp.reward
+        # repeats might be zero for StepType.FIRST
+        rl_info, repeats, sample_rewards = rl_exp.rollout_info
+
+        if self._reward_normalizer is not None:
+            normalizer = self._reward_normalizer._normalizer
+            normalizer.update(sample_rewards)
+
+            # compute current variance
+            m = normalizer._mean_averager.get()
+            m2 = normalizer._m2_averager.get()
+            var = torch.relu(m2 - m**2)
+
+            # compute accumulated mean over ``repeats`` steps
+            acc_mean = ((1 - torch.pow(self._gamma, repeats.to(torch.float32)))
+                        / (1 - self._gamma) * m)
+
+            reward -= acc_mean
```

Comment:
why not do this using normalize_along_batch_dims?

Reply:
acc_mean has a batch size > 1 in this case, which is not allowed by normalize_along_batch_dims